src/Chapter2.hs

```diff
@@ -842,7 +854,11 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [a] -> [a]
+rotate _ [] = []
+rotate x xs = drop x $ take len (cycle xs)
```

Comment:
That's great!

Another optimisation could be using `mod` by the length of the list on `x`, so that you won't do additional unnecessary work 🙂